### PR TITLE
Implement MobX

### PR DESCRIPTION
**Pull request by samcooke98**
I noticed the mobX tests went walkabout. I added them back, mainly cause I was interested to see how MobX goes.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -18,7 +18,11 @@ document.title = name;
 
 // concurrent mode
 const root = ReactDOM.createRoot(document.getElementById('app'));
-root.render(<App />);
+root.render(
+  <React.StrictMode>
```
Comment (samcooke98) [inline src/index.js:22]: The React Profiler was complaining about StrictMode being missing, so I added it here. Not sure if you would prefer this in a separate PR.

**Comment by dai-shi** [comment]
Hi, the reason why mobx is removed is because the impl didn't use the common reducer. Can you try something like valtio?

**Comment by samcooke98** [comment]
> Hi, the reason why mobx is removed is because the impl didn't use the common reducer. Can you try something like valtio?

Ah interesting. 

> Can you try something like valtio?

The reason I wanted to introduce MobX is that it's the chose state management library at the company I work for. 

I could maybe make it try and use the reducer?

**Comment by dai-shi** [comment]
Yeah, it would be more maintainable if we were to merge this.
Otherwise, feel free to run all tests on your end and update the results. (Leaving this PR open for reference.)

**Comment by kubk** [comment]
Is it going to be merged? Why the impl have to use a common reducer?

**Comment by dai-shi** [comment]
Because we might be changing the common reducer in the future. Would you like to give it a try?

**Comment by kubk** [comment]
@dai-shi Thanks, it's clear 👍 

@samcooke98 Hi, are you going change the implementation to use reducer?

**Comment by dai-shi** [comment]
@kubk Feel free to work on it (please open a new PR).

Closing as stale.

**Pull request closed without merging**